.buildkite/pipeline-e2e-tests.yml

```diff
@@ -20,12 +20,10 @@ steps:
           image: docker.elastic.co/ci-agent-images/cloud-k8s-operator/buildkite-agent:ef29aa4e
           memory: "2G"
 
-      # for all tags
       # for nightly builds from main
       - label: ":buildkite: e2e"
         if: |
-          build.tag != null
-          || ( build.branch == "main" && build.source == "schedule" )
+          ( build.branch == "main" && build.source == "schedule" )
           || build.message =~ /^buildkite test .*e2e\/all.*/
```

Comment:
To give some context on why I was suggesting removing it, I've only used it once and we mainly use PR comments to trigger given e2e tests.  But to clean this properly, it would be more appropriate to do it in a separate PR. In the end, it doesn't bother, if it's used (not sure?), we have to keep it.